scarb/src/core/publishing/manifest_normalization.rs

```diff
@@ -0,0 +1,146 @@
+    TomlWorkspaceField,
+};
+
+pub fn prepare_manifest_for_publish(pkg: &Package) -> Result<TomlManifest> {
+    let package = Some(generate_package(pkg));
+
+    let dependencies = Some(generate_dependencies(
+        // NOTE: We deliberately do not ask for `full_dependencies` here, because
+        // we do not want to emit requirements for built-in packages like `core`.
+        &pkg.manifest.summary.dependencies,
+        DepKind::Normal,
+    )?);
+
+    let tool = pkg.manifest.metadata.tool_metadata.clone().map(|m| {
+        m.into_iter()
+            .map(|(k, v)| (k, MaybeWorkspace::Defined(v)))
+            .collect()
+    });
+
+    Ok(TomlManifest {
+        package,
+        workspace: None,
+        dependencies,
+        lib: None,
+        // TODO(mkaput): Allow packaging Cairo plugins.
+        cairo_plugin: None,
+        test: None,
+        target: None,
+        cairo: None,
+        profile: None,
+        scripts: None,
+        tool,
+    })
+}
+
+fn generate_package(pkg: &Package) -> Box<TomlPackage> {
+    let summary = &pkg.manifest.summary;
+    let metadata = &pkg.manifest.metadata;
+    Box::new(TomlPackage {
+        name: summary.package_id.name.clone(),
+        version: MaybeWorkspace::Defined(summary.package_id.version.clone()),
+        authors: metadata.authors.clone().map(MaybeWorkspace::Defined),
+        urls: metadata.urls.clone(),
+        description: metadata.description.clone().map(MaybeWorkspace::Defined),
+        documentation: metadata.documentation.clone().map(MaybeWorkspace::Defined),
+        homepage: metadata.homepage.clone().map(MaybeWorkspace::Defined),
+        keywords: metadata.keywords.clone().map(MaybeWorkspace::Defined),
+        license: metadata.license.clone().map(MaybeWorkspace::Defined),
+        // TODO(mkaput): Normalize this the same way as readme is.
+        license_file: metadata.license_file.clone().map(MaybeWorkspace::Defined),
+        readme: metadata
+            .readme
+            .as_ref()
+            .map(|p| map_metadata_file_path(p, pkg)),
+        repository: metadata.repository.clone().map(MaybeWorkspace::Defined),
+        no_core: summary.no_core.then_some(true),
+        cairo_version: metadata.cairo_version.clone().map(MaybeWorkspace::Defined),
+    })
+}
+
+fn generate_dependencies(
+    deps: &[ManifestDependency],
+    kind: DepKind,
+) -> Result<BTreeMap<PackageName, MaybeWorkspace<TomlDependency, TomlWorkspaceDependency>>> {
+    deps.iter()
+        .filter(|dep| dep.kind == kind)
+        .map(|dep| {
+            let name = dep.name.clone();
+            let toml_dep = generate_dependency(dep)?;
+            Ok((name, MaybeWorkspace::Defined(toml_dep)))
+        })
+        .collect()
+}
+
+fn generate_dependency(dep: &ManifestDependency) -> Result<TomlDependency> {
+    assert!(
+        !dep.source_id.is_std(),
+        "Built-in dependencies should not be included in published manifests."
```

Comment:
Do I understand correctly, that `"starknet"` will not trigger this assert because it is registry dep until patched? Would it make sense to add it to one of the tests to make sure?

Reply:
It is tested: https://github.com/software-mansion/scarb/blob/e209612a1f8ed85a186720c8435c5fba49ff6c71/scarb/tests/package.rs#L325-L326